librabbitmq/amqp_openssl.c

```diff
@@ -662,8 +663,22 @@ destroy_openssl(void)
      * safely. We do leak the amqp_openssl_lockarray. Which is only
      * an issue if you repeatedly unload and load the library
      */
+    ERR_remove_state(0);
+    FIPS_mode_set(0);
     CRYPTO_set_locking_callback(NULL);
     CRYPTO_set_id_callback(NULL);
+    ENGINE_cleanup();
+    CONF_modules_free();
+    EVP_cleanup();
+    CRYPTO_cleanup_all_ex_data();
+    ERR_free_strings();
+    free(amqp_openssl_lockarray);
+    amqp_openssl_lockarray = NULL;
+    /* Allow the possibility to reinitialze openSSL again */
+    openssl_initialized = 0;
+    /* In newer versions of OpenSSL, call:
+     SSL_COMP_free_compression_methods();
```

Comment:
Probably need to hack this up with `#ifdef`'s depending on what version of OpenSSL is being compiled against.

See: ~do_init() in http://www.boost.org/doc/libs/1_64_0/boost/asio/ssl/detail/impl/openssl_init.ipp